### PR TITLE
Feature/library details view

### PR DESCRIPTION
**Pull request by soy-park**
### Types of changes made?
- [ ]  documentation
- [ ]  testing
- [x]  functionality
- [ ]  styling
- [x]  refactor
- [ ]  bug fix

### What does this PR do?
- Adds functionality: I created a LibraryDetailsView React component that contains selectors to access the states of libraryDetails and libraryIndex, dispatches fetchBooks, and renders the library's name, address, and list of books, as well as a button to add a book. The button is a link to a form page. 

### Relevant Tickets?
- Finishes iteration or ticket for:  LibraryDetails page view

### Issues / Further review needed?
- I created a getLibrary() that takes in an id as a parameter to find a specific library that matches the id parameter. In order to instantiate the function to render details of a specific library details, I had to add an argument (i.e. specific id), so I used "1" as a placeholder. May need to change this to make it more dynamic. 

### Questions / Comments / Relevant Screenshots?
- Installed React Router v5
- Imported Router and NavLink in App

### CODE REVIEW

File: src/react-components/libraryDetailsView.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { useAppSelector, useAppDispatch } from '../redux/store';
+import { fetchBooks } from '../redux/libraryDetails';
+import { Link } from 'react-router-dom';
+
+export const LibraryDetailsView = () => {
+  const libraryDetails = useAppSelector(state => state.libraryDetails);
+  const libraryIndex = useAppSelector(state => state.libraryIndex);
+  const dispatch = useAppDispatch();
+
+  const getLibrary = (id: number) => {
```
Comment (tialaaa) [inline src/react-components/libraryDetailsView.tsx:11]: Lines 11-15 make total sense for development. When we incorporate this page's Route into `App.tsx` these could move into that Route element. And then we could use `match.params` within that Route to get the dynamic library id.

File: src/redux/libraryIndex.ts [hunk 2]
```diff
@@ -25,7 +25,8 @@ interface Library {
         address: {
             street: string, 
             city: string, 
-            state: string
+            state: string,
+            zip: number
```
Comment (tialaaa) [inline src/redux/libraryIndex.ts:29]: Good catch on the missing Zip code! That wasn't included in our mock server sample data.

**Comment by tialaaa** [comment]
Looks good! We can address your issue callout with our Routes as we go forward.

File: src/react-components/libraryDetailsView.tsx (hunk 1 above)
Comment (soy-park) [reply src/react-components/libraryDetailsView.tsx:11]: Yes, sounds good

**Comment by tialaaa** [comment]
Nice work collaborating on the merge conflicts.

**Pull request merged**